classes/operations/label_manager.rb

```diff
@@ -0,0 +1,59 @@
+require_relative '../structure/label'
+require_relative '../../module/data_operations'
+
+class LabelManager
+  include DataOperations
+
+  attr_accessor :label_data
+
+  def initialize
+    @label_data = fetch_data('label')
+  end
+
+  def create_label
+    state = false
+
+    until state
+
+      puts 'Please enter the label title:'
+      title = gets.chomp.to_s
+      state = true unless title == ''
+      puts 'Label title Required' unless state == true
+      title
+    end
+
+    puts 'Enter the color of the label'
+    color = gets.chomp
+
+    new_label = Label.new(title, color)
+
+    @label_data << new_label.to_hash
+
+    update_data('label', new_label.to_hash)
+    new_label
+  end
+
+  def list_labels
+    @label_data.each_with_index do |label, index|
+      puts '---------------------'
+      puts "(#{index}) #{label['id']} - #{label['title']} - #{label['color']})"
+      puts '---------------------'
+    end
+
+    puts "(#{@label_data.length}) - Create a new label"
+  end
+
+  def format_label(hashed_label)
+    label = Label.new(hashed_label['title'])
+    label.id = hashed_label['id']
```

Comment:
I also noticed that this line gives an error because the `id` attribute in the label class can only be read and not the written to.

Consider removing the `id` property from the `attr_reader` and put in `attr_accessor :id`, this way you will be able to set to the class property

Reply:
@clintonjosephs you are right, I did not check if a book could be added by selecting an existing label, now it is fixed, thanks for mentioning that